ArjunaJTA/cdi/classes/com/arjuna/ats/jta/cdi/NarayanaTransactionManager.java

```diff
@@ -120,92 +131,46 @@
    * @see TransactionScoped
    */
   @Inject
-  NarayanaTransactionManager(final BeanManager beanManager,
+  NarayanaTransactionManager(final Instance<JTAEnvironmentBean> jtaEnvironmentBeanInstance,
                              @Initialized(TransactionScoped.class)
                              final Event<Transaction> transactionScopeInitializedBroadcaster,
                              @Destroyed(TransactionScoped.class)
                              final Event<Object> transactionScopeDestroyedBroadcaster) {
-    super(getDelegate(beanManager));
+    super(jtaEnvironmentBeanInstance == null ? null : getDelegate(jtaEnvironmentBeanInstance));
     this.transactionScopeInitializedBroadcaster = transactionScopeInitializedBroadcaster;
     this.transactionScopeDestroyedBroadcaster = transactionScopeDestroyedBroadcaster;
   }
 
-  private static final TransactionManager getDelegate(final BeanManager beanManager) {
+  /**
+   * Returns a {@link TransactionManager} to use as a delegate {@link
+   * TransactionManager} for a {@link NarayanaTransactionManager}
+   * instance.
+   *
+   * <p>This method may return {@code null}.</p>
+   *
+   * @param jtaEnvironmentBeanInstance an {@link Instance} capable of
+   * supplying a {@link JTAEnvironmentBean}; may be {@code null}
+   *
+   * @return a {@link TransactionManager}, or {@code null}
+   *
+   * @exception CreationException if an error occurs
+   *
+   * @see JTASupplier#get(String, Supplier)
+   */
+  private static final TransactionManager getDelegate(final Instance<JTAEnvironmentBean> jtaEnvironmentBeanInstance) {
+    final JTAEnvironmentBean jtaEnvironmentBean;
+    if (jtaEnvironmentBeanInstance == null) {
+      jtaEnvironmentBean = BeanPopulator.getDefaultInstance(JTAEnvironmentBean.class);
+    } else {
+      jtaEnvironmentBean = jtaEnvironmentBeanInstance.get();
```

Comment:
I think there could be a wrong assumption that the `JTAEnvironmentBean` is the CDI bean. It's just a pojo class instantiated dynamically by Narayana internal class loading utilities. When I run in SE environment the `jaEnvironmentBeanInstance` is not `null` but it's `@Default Instance<JTAEnvironmentBean>` which is not backed by a bean (I expect) and the call ends with error[1]. The `JTAEnvironmentBean` should be gathered like `jtaPropertyManager.getJTAEnvironmentBean()` (I assume)

[1]
```
INFO: WELD-ENV-002003: Weld SE container f87da1e4-a926-4038-a034-44152d8745a6 initialized
Tests run: 2, Failures: 0, Errors: 2, Skipped: 0, Time elapsed: 0.689 sec <<< FAILURE!
testTransactionScoped(org.jboss.narayana.quickstarts.jta.CDIBindingTestCase)  Time elapsed: 0.648 sec  <<< ERROR!
org.jboss.weld.exceptions.UnsatisfiedResolutionException: WELD-001334: Unsatisfied dependencies for type JTAEnvironmentBean with qualifiers @Default 
 at org.jboss.weld.bean.builtin.InstanceImpl.checkBeanResolved(InstanceImpl.java:241)
 at org.jboss.weld.bean.builtin.InstanceImpl.get(InstanceImpl.java:113)
 at com.arjuna.ats.jta.cdi.NarayanaTransactionManager.getDelegate(NarayanaTransactionManager.java:165)
 at com.arjuna.ats.jta.cdi.NarayanaTransactionManager.<init>(NarayanaTransactionManager.java:139)
 at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
 at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
 at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
 at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
 at org.jboss.weld.injection.ConstructorInjectionPoint.newInstance(ConstructorInjectionPoint.java:119)
 at org.jboss.weld.injection.ConstructorInjectionPoint.invokeAroundConstructCallbacks(ConstructorInjectionPoint.java:92)
 at org.jboss.weld.injection.ConstructorInjectionPoint.newInstance(ConstructorInjectionPoint.java:78)
 at org.jboss.weld.injection.producer.AbstractInstantiator.newInstance(AbstractInstantiator.java:28)
 at org.jboss.weld.injection.producer.BasicInjectionTarget.produce(BasicInjectionTarget.java:112)
 at org.jboss.weld.injection.producer.BeanInjectionTarget.produce(BeanInjectionTarget.java:186)
 at org.jboss.weld.bean.ManagedBean.create(ManagedBean.java:158)
 at org.jboss.weld.contexts.AbstractContext.get(AbstractContext.java:96)
 at org.jboss.weld.bean.ContextualInstanceStrategy$DefaultContextualInstanceStrategy.get(ContextualInstanceStrategy.java:100)
 at org.jboss.weld.bean.ContextualInstanceStrategy$ApplicationScopedContextualInstanceStrategy.get(ContextualInstanceStrategy.java:140)
 at org.jboss.weld.bean.ContextualInstance.get(ContextualInstance.java:50)
 at org.jboss.weld.bean.proxy.ContextBeanInstance.getInstance(ContextBeanInstance.java:102)
 at org.jboss.weld.bean.proxy.ProxyMethodHandler.getInstance(ProxyMethodHandler.java:131)
 at com.arjuna.ats.jta.cdi.NarayanaTransactionManager$Proxy$_$$_WeldClientProxy.begin(Unknown Source)
 at org.jboss.narayana.quickstarts.jta.CDIBindingTestCase.testTransactionScoped(CDIBindingTestCase.java:78)
```

Reply:
Oops; I simply left out a `jtaEnvironmentBeanInstance.isUnsatisfied()` call; will fix!

Reply:
Should be fixed.